soc/arm/st_stm32/stm32l1/Kconfig.defconfig.stm32l151xb

```diff
@@ -9,7 +9,7 @@ if SOC_STM32L151XB
 
 config SOC
 	string
-	default "stm32l151xb"
+	default "stm32l151xba"
```

Comment:
This can't be right, there are both xxxxB and xxxxBA parts in the wild.  See https://www.st.com/en/microcontrollers-microprocessors/stm32l151-152.html  They're both even still marked as ACTIVE.

See https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-ultra-low-power-mcus/stm32l1-series/stm32l151-152/stm32l151rb.html vs https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-ultra-low-power-mcus/stm32l1-series/stm32l151-152/stm32l151rb-a.html

Reply:
> This can't be right, there are both xxxxB and xxxxBA parts in the wild.

I didn't say that `stm32l151xb` is deprecated. I assumed that in zephyr mainline, there is only one L1X SoC present and it is based on the Wistrio board. So the exact part number of the SoC is found to be `stm32l151xba`, hence I just modified it.

@erwango What is the criteria for `default` SoC name?

Reply:
> this isn't the default soc, the file you're editing here is "Kconfig.defconfig.stm32l151xb" you're changing it to add an "a" suffix.

I think I misunderstood the actual difference between those 2 SoCs. Anyway, I'v asked the board vendor to clarify the SoC part number as there seems to be different versions of RAK811 modules. If they confirm it is `STM32L151XB-A` then I'll do the change accordingly.

Reply:
The `stm32flash` utility confirms that it is `STM32L151XB-A`.

Reply:
> @erwango What is the criteria for default SoC name?

So, first it should match your SoC. Then, it should match (in lower case), the definition in `stm32cube/stm32l1xx/soc/stm32l1xx.h`.
It will be used in `stm32cube/CMakeLists.txt`

Reply:
@erwango So for the `STM32L151xBA` SoC, I should create `soc/arm/st_stm32/stm32l1/Kconfig.defconfig.stm32l151xba`, right? It will be an exact duplicate of `stm32l151xb` though.

Reply:
Yes, this is needed, but if I'm correct, @karlp did it already

Reply:
I actually added the x8-a, because that's the part I wanted for something else.  I raised it on the slack channel, what the future plans were for collecting all the variants, because you're goign to explode in boilerplate. response there was "just add them as you need them, we'll sort it out later"

The _defconfig_ might be, but it will refer to a different dtsi file, as they have different flash/ram to the non-a parts.  (and less eeprom writes too, but that's getting off topic...)

Reply:
> Yes, this is needed, but if I'm correct, @karlp did it already

@erwango He added `x8-a` but for me, `xb-a` is needed. So I'll create a new one!